mapper/orm/src/main/java/org/hibernate/search/mapper/orm/loading/impl/HibernateOrmJpaMassIndexingTypeLoadingStrategy.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.loading.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.persistence.LockModeType;
+import org.hibernate.CacheMode;
+import org.hibernate.FlushMode;
+import org.hibernate.ScrollMode;
+import org.hibernate.ScrollableResults;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.persister.entity.EntityPersister;
+import org.hibernate.query.Query;
+import org.hibernate.search.mapper.orm.common.impl.HibernateOrmUtils;
+import org.hibernate.search.mapper.orm.logging.impl.Log;
+import org.hibernate.search.mapper.orm.massindexing.impl.HibernateOrmMassIndexingMappingContext;
+import org.hibernate.search.mapper.orm.session.impl.HibernateOrmSessionTypeContextProvider;
+import org.hibernate.search.mapper.pojo.loading.EntityIdentifierScroll;
+import org.hibernate.search.mapper.pojo.loading.EntityLoader;
+import org.hibernate.search.mapper.pojo.loading.EntityLoadingTypeGroup.GroupingType;
+import org.hibernate.search.mapper.pojo.massindexing.loader.MassIndexingEntityLoadingStrategy;
+import org.hibernate.search.mapper.pojo.massindexing.loader.MassIndexingThreadContext;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+import org.hibernate.search.mapper.pojo.loading.EntityLoadingTypeGroup;
+
+public class HibernateOrmJpaMassIndexingTypeLoadingStrategy<E, I> implements
```

Comment:
It's been a long time since I noticed that the document loading part is similar to the "EntityLoader" in loading entities during a search. But that would require looking for a common solution and also using the EntityLoader with mapper -ojeo to load entities during the search. Can it be done including PR?

Reply:
To be clear, I'm not suggestion to merge the code that load entities during search with the code that loads entities during mass indexing. That's clearly out of scope.

What I'm saying is that before your PR, the `EntityLoadingStrategy` was creating the loader for mass indexing. Now it doesn't anymore. There's something wrong, and I suggest you just move the code of `HibernateOrmJpaMassIndexingTypeLoadingStrategy` back to `EntityLoadingStrategy`.

No need to change anything to entity loading during search.

Reply:
There was a problem that it uses an element from SPI, PojoLoader. The mapper-javabean is called in "LoadingOptions" from the API. It would be difficult to develop a common interface here. But after the changes, you can use the WpoLoader inside the API by just extracting "EntityIdentifierScroll".  

I will try to remove EntityLoader and check how to combine the strategy in hibernate-orm. I will also consider how to find a common solution for mapper-javabean.

Reply:
To extend EntityLoadingStrategy I had to rename it to "HibernateOrmEntityLoadingStrategy". These are the rules for style checking. I added a new type "AbstractHibernateOrmLoadingStrategy" implementing "MassIndexingEntityLoadingStrategy". Now "HibernateOrmNonEntityIdPropertyEntityLoadingStrategy" and "HibernateOrmEntityIdEntityLoadingStrategy" extend the "HibernateOrmNonEntityIdPropertyEntityLoadingStrategy" type. Thanks to this I do not have to duplicate functions implementing "MassIndexingEntityLoadingStrategy". Now the same EntityLoadingStrategy handles entity loading and indexing as suggested.